coriolis/tests/scheduler/rpc/test_client.py

```diff
@@ -0,0 +1,221 @@
+            mock_get_workers.return_value = [{'id': 'test_id'}]
+
+            self.client.get_worker_service_for_task(
+                mock.sentinel.ctxt, self.task, self.origin_endpoint,
+                self.destination_endpoint)
+
+    @mock.patch.object(client.SchedulerClient, 'get_workers_for_specs')
+    @mock.patch.object(tasks_factory, 'get_task_runner_class')
+    def test_get_worker_service_for_task_different_provider_types(
+            self, mock_get_task_runner_class, mock_get_workers):
+        for provider_type in [constants.PROVIDER_PLATFORM_SOURCE,
+                              constants.PROVIDER_PLATFORM_DESTINATION]:
+            mock_get_task_runner_class.return_value.get_required_platform.\
+                return_value = constants.TASK_PLATFORM_SOURCE
+            mock_get_task_runner_class.return_value.\
+                get_required_provider_types.return_value = {
+                    provider_type: 'type'}
+            mock_get_workers.return_value = [{'id': 'test_id'}]
+
+            self.client.get_worker_service_for_task(
+                mock.sentinel.ctxt, self.task, self.origin_endpoint,
+                self.destination_endpoint)
+
+    @mock.patch.object(client.SchedulerClient, 'get_workers_for_specs')
+    @mock.patch.object(tasks_factory, 'get_task_runner_class')
+    def test_get_worker_service_for_task_retry(
+            self, mock_get_task_runner_class, mock_get_workers):
+        mock_get_task_runner_class.return_value.get_required_platform.\
+            return_value = constants.TASK_PLATFORM_SOURCE
+        mock_get_task_runner_class.return_value.get_required_provider_types.\
+            return_value = {constants.PROVIDER_PLATFORM_SOURCE:
+                            'provider_type'}
+        mock_get_workers.side_effect = [Exception(), [{'id': 'test_id'}]]
+
+        with self.assertLogs('coriolis.scheduler.rpc.client',
+                             level=logging.WARN):
+            self.client.get_worker_service_for_task(
+                mock.sentinel.ctxt, self.task, self.origin_endpoint,
+                self.destination_endpoint)
+
+    @mock.patch.object(client.SchedulerClient, 'get_workers_for_specs')
+    @mock.patch.object(tasks_factory, 'get_task_runner_class')
+    def test_get_worker_service_for_task_no_suitable_worker(
+            self, mock_get_task_runner_class, mock_get_workers):
+        mock_get_task_runner_class.return_value.get_required_platform.\
+            return_value = constants.TASK_PLATFORM_SOURCE
+        mock_get_task_runner_class.return_value.get_required_provider_types.\
+            return_value = {constants.PROVIDER_PLATFORM_SOURCE: 'type'}
+        mock_get_workers.side_effect = Exception()
+
+        self.assertRaises(
+            exception.NoSuitableWorkerServiceError,
+            self.client.get_worker_service_for_task, mock.sentinel.ctxt,
+            self.task, self.origin_endpoint, self.destination_endpoint)
```

Comment:
You could also set `retry_period=0` to make it run faster. You could apply this anywhere.